main.go

```diff
@@ -12,118 +12,75 @@ import (
-	if err != nil {
-		panic(microerror.JSON(err))
-	}
-
-	awsmachinedeploymentValidator, err := awsmachinedeployment.NewValidator(config)
-	if err != nil {
-		panic(microerror.JSON(err))
+		_, _ = fmt.Fprintf(os.Stderr, microerror.Pretty(err, true))
 	}
+}
 
-	clusterValidator, err := cluster.NewValidator(config)
+func mainErr() error {
+	webhookConfig, err := config.Parse()
 	if err != nil {
-		panic(microerror.JSON(err))
+		return microerror.Mask(err)
 	}
 
-	g8scontrolplaneValidator, err := g8scontrolplane.NewValidator(config)
-	if err != nil {
-		panic(microerror.JSON(err))
+	scheme := runtime.NewScheme()
+	{
+		err = v1alpha3.AddToScheme(scheme)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		err = v1alpha4.AddToScheme(scheme)
+		if err != nil {
+			return microerror.Mask(err)
+		}
 	}
 
-	machinedeploymentValidator, err := machinedeployment.NewValidator(config)
-	if err != nil {
-		panic(microerror.JSON(err))
-	}
+	var crdHandler http.Handler
+	{
+		crdConverter, err := crdconverter.New(crdconverter.Config{
+			Logger: webhookConfig.Logger,
+		})
+		if err != nil {
+			return microerror.Mask(err)
+		}
 
-	networkPoolValidator, err := networkpool.NewValidator(config)
-	if err != nil {
-		panic(microerror.JSON(err))
+		crdHandler, err = crdhandler.New(crdhandler.Config{
+			Logger:    webhookConfig.Logger,
+			Converter: crdConverter,
+			Scheme:    scheme,
+		})
+		if err != nil {
+			return microerror.Mask(err)
+		}
 	}
 
 	// Here we register our endpoints.
 	handler := http.NewServeMux()
-	handler.Handle("/mutate/awscluster", mutator.Handler(awsclusterMutator))
-	handler.Handle("/mutate/awsmachinedeployment", mutator.Handler(awsmachinedeploymentMutator))
-	handler.Handle("/mutate/awscontrolplane", mutator.Handler(awscontrolplaneMutator))
-	handler.Handle("/mutate/cluster", mutator.Handler(clusterMutator))
-	handler.Handle("/mutate/g8scontrolplane", mutator.Handler(g8scontrolplaneMutator))
-	handler.Handle("/mutate/machinedeployment", mutator.Handler(machinedeploymentMutator))
-	handler.Handle("/validate/awscluster", validator.Handler(awsclusterValidator))
-	handler.Handle("/validate/awscontrolplane", validator.Handler(awscontrolplaneValidator))
-	handler.Handle("/validate/awsmachinedeployment", validator.Handler(awsmachinedeploymentValidator))
-	handler.Handle("/validate/cluster", validator.Handler(clusterValidator))
-	handler.Handle("/validate/g8scontrolplane", validator.Handler(g8scontrolplaneValidator))
-	handler.Handle("/validate/machinedeployment", validator.Handler(machinedeploymentValidator))
-	handler.Handle("/validate/networkpool", validator.Handler(networkPoolValidator))
+	handler.Handle("/crdconvert", crdHandler)
 
 	handler.HandleFunc("/healthz", healthCheck)
 
 	metrics := http.NewServeMux()
 	metrics.Handle("/metrics", promhttp.Handler())
 
-	go serveMetrics(config, metrics)
-	serveTLS(config, handler)
+	go serveMetrics(webhookConfig, metrics)
+	http.ListenAndServe(":8443", handler)
```

Comment:
Disabling TLS for testing temporarily. Need to reenable for Kubernetes to be able to talk to it.